crates/proof-of-sql/src/sql/parse/query_expr_tests.rs

```diff
@@ -1596,107 +1599,99 @@ fn we_can_parse_arithmetic_expression_within_aggregations_in_the_result_expr() {
 
 #[test]
 fn we_cannot_use_non_grouped_columns_outside_agg() {
-    assert!(matches!(
-        query!(select: ["i"], group: ["s"], should_err: true),
-        ConversionError::PostprocessingError {
-            source: PostprocessingError::IdentifierNotInAggregationOperatorOrGroupByClause { .. }
-        }
-    ));
-    assert!(matches!(
-        query!(select: ["sum(i)", "i"], group: ["s"], should_err: true),
-        ConversionError::PostprocessingError {
-            source: PostprocessingError::IdentifierNotInAggregationOperatorOrGroupByClause { .. }
-        }
-    ));
-    assert!(matches!(
-        query!(select: ["min(i) + i"], group: ["s"], should_err: true),
-        ConversionError::PostprocessingError {
-            source: PostprocessingError::IdentifierNotInAggregationOperatorOrGroupByClause { .. }
-        }
-    ));
-    assert!(matches!(
-        query!(select: ["2 * i", "min(i)"], group: ["s"], should_err: true),
-        ConversionError::PostprocessingError {
-            source: PostprocessingError::IdentifierNotInAggregationOperatorOrGroupByClause { .. }
-        }
-    ));
-    assert!(matches!(
-        query!(select: ["2 * i", "min(i)"], should_err: true),
-        ConversionError::InvalidGroupByColumnRef { .. }
-    ));
-    assert!(matches!(
-        query!(select: ["sum(i)", "i"], should_err: true),
-        ConversionError::InvalidGroupByColumnRef { .. }
-    ));
-    assert!(matches!(
-        query!(select: ["max(i) + 2 * i"], should_err: true),
-        ConversionError::InvalidGroupByColumnRef { .. }
-    ));
+    let query_texts = vec![
+        "select i from sxt.t group by s",
+        "select sum(i), i from sxt.t group by s",
+        "select min(i) + i from sxt.t group by s",
+        "select 2 * i, min(i) from sxt.t group by s",
+        "select 2 * i, min(i) from sxt.t",
+        "select sum(i), i from sxt.t",
+        "select max(i) + 2 * i from sxt.t",
+    ];
+
+    for query_text in query_texts {
+        let intermediate_ast = SelectStatementParser::new().parse(query_text).unwrap();
+        let result = QueryExpr::<NaiveCommitment>::try_new(intermediate_ast, None, &None);
+
+        assert!(matches!(
+            result.unwrap_err(),
+            ConversionError::PostprocessingError {
+                source: PostprocessingError::IdentifierNotInAggregationOperatorOrGroupByClause { .. }
+            }
+        ) || matches!(
+            result.unwrap_err(),
+            ConversionError::InvalidGroupByColumnRef { .. }
+        ));
+    }
```

Comment:
Can you make this two for loops instead of one so that we can distinguish between the two error conditions and ensure we are getting the expected one?

(This isn't the only place where this applies, so be sure to do the same thing elsewhere as well.)